blockchain/chain_sync/src/sync_state.rs

```diff
@@ -0,0 +1,133 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use blocks::{tipset::tipset_json::TipsetJsonRef, Tipset};
+use clock::ChainEpoch;
+use serde::{Serialize, Serializer};
+use std::fmt;
+use std::sync::Arc;
+use std::time::SystemTime;
+
+/// Current state of the ChainSyncer using the BlockSync protocol.
+#[derive(PartialEq, Debug, Clone, Copy)]
+pub enum SyncStage {
+    /// Syncing headers from the heaviest tipset to genesis.
+    Headers,
+    /// Persisting headers on chain from heaviest to genesis.
+    PersistHeaders,
+    /// Syncing messages and performing state transitions.
+    Messages,
+    /// ChainSync completed and is following chain.
+    Complete,
+    /// Error has occured while syncing.
+    Errored,
```

Comment:
I think Error is sufficient, i dont think errored is a word BUT language is what we make it hahaha